<launch>

    <!-- common -->
    <param name="robot_description" command="$(find xacro)/xacro.py $(find ros_arm_moveit)/urdf/mycobot_urdf.xacro" />

    <!-- ***********************moveit***************************** -->

    <rosparam file="$(find ros_arm_moveit)/config/controllers.yaml" command="load" />
    <!-- aibot_arm_real_node -->

    <node pkg="ros_arm_moveit" type="arm_real_node" name="arm_real_node" output="screen">
    </node>

    <node pkg="ros_arm_moveit" type="mycobot_control" name="mycobot_control" output="screen">
            <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
            <param name="serial_baudrate"     type="int"    value="115200"/>
    </node>


    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/" args="
            /aibot/controller/joint_state_controller
            /aibot/controller/mycobot_controller
            "/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="publish_frequency" type="double" value="100.0" />
    </node>

    <include file="$(find mycobot_moveit_config)/launch/move_group.launch" 
    >
        <arg name="allow_trajectory_execution" value="true"/>
        <arg name="fake_execution" value="false"/>
        <arg name="info" value="true"/>
        <param name="allowed_execution_duration_scaling" value="5"/>
        <param name="allowed_goal_duration_margin" value="5"/>
        <param name="execution_duration_monitoring" value="false"/>
        <arg name="publish_monitored_planning_scene" value="true" />
    </include>

    <!-- ***********************moveit***************************** -->


    <node name="rviz" type="rviz" pkg="rviz" args="-d $(find ros_arm_moveit)/launch/rviz/mycobot_arm_moveit.rviz" />


</launch>

